src/hotspot/cpu/x86/x86.ad

```diff
@@ -8269,6 +8598,569 @@ instruct vmask_first_or_last_true_avx(rRegI dst, vec mask, rRegL tmp, vec xtmp,
 %}
 #endif // _LP64
 
+// ---------------------------------- Vector Masked Operations ------------------------------------
+
+instruct vadd_reg_masked(vec dst, vec src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (AddVB (Binary dst src2) mask));
+  match(Set dst (AddVS (Binary dst src2) mask));
+  match(Set dst (AddVI (Binary dst src2) mask));
+  match(Set dst (AddVL (Binary dst src2) mask));
+  match(Set dst (AddVF (Binary dst src2) mask));
+  match(Set dst (AddVD (Binary dst src2) mask));
```

Comment:
Could we add the match rules for sub, and, or, xor, mul, div, min, max, rearrange here?
The ins_encode part it the same for all of these.

Reply:
We will not be saving anything by clubbing the rules,  on pattern per operation is easy to review and maintain.
Your suggestion will make one pattern very bulky.